### PR TITLE
dot11 ack frame dissection

### PR DESCRIPTION
**Pull request by haim0n**

### CODE REVIEW

File: scapy/layers/dot11.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def mysummary(self):
     def guess_payload_class(self, payload):
         if self.type == 0x02 and (0x08 <= self.subtype <= 0xF and self.subtype != 0xD):
             return Dot11QoS
-	elif self.FCfield & 0x40:
+        elif self.FCfield & 0x40:
```
Comment (p-l-) [inline scapy/layers/dot11.py]: Hi, please do not alter the lines you don't actually modify, since this breaks the code history.

**Comment by guedou** [comment]
@haim0n is your patch complete ?

**Comment by haim0n** [comment]
@guedoe,
Yep. It's complete.

**Comment by guedou** [comment]
Thanks for your answer. It seems that I don't understand your patch =)

Could you provide a use case and a unit test ? This might help me.

**Comment by guedou** [comment]
From https://www.cloudshark.org/captures/f7cfe4f0a897 I managed to find the following Ack frame:

```python
raw = '\x00\x00\x1a\x00/H\x00\x00=\x06\x99\x00\x00\x00\x00\x00\x10\x0b\x9e\t\xa0\x00\xad\x02\x00\x00\xd4\x00\x00\x00\xe4\xce\x8f\xb3/#P\xe7\xd9\xac'
```

Without you patch, Scapy displays:
```python
>>> RadioTap(raw)
<RadioTap  version=0 pad=0 len=26 present=TSFT+Flags+Rate+Channel+dBm_AntSignal+Antenna+b14 notdecoded='=\x06\x99\x00\x00\x00\x00\x00\x10\x0b\x9e\t\xa0\x00\xad\x02\x00\x00' |<Dot11  subtype=13L type=Control proto=0L FCfield= ID=0 addr1=e4:ce:8f:b3:2f:23 addr2=None addr3=None SC=None addr4=None |<Raw  load='P\xe7\xd9\xac' |>>>
```

With your patch:
```python
>>> RadioTap(raw)
<RadioTap  version=0 pad=0 len=26 present=TSFT+Flags+Rate+Channel+dBm_AntSignal+Antenna+b14 notdecoded='=\x06\x99\x00\x00\x00\x00\x00\x10\x0b\x9e\t\xa0\x00\xad\x02\x00\x00' |<Dot11  subtype=13L type=Control proto=0L FCfield= ID=0 addr1=e4:ce:8f:b3:2f:23 addr2=None addr3=None SC=None addr4=None |<Dot11Ack  |<Raw  load='P\xe7\xd9\xac' |>>>>
```

**Comment by haim0n** [comment]
@guedou ,

I was going to provide some UT, honestly :)

**Comment by p-l-** [comment]
@haim0n you can still do so, that would be nice!

**Comment by haim0n** [comment]
@p-l- ,

will do - I just need some quality time with the uts framework.

**Comment by p-l-** [comment]
@haim0n great!

**Comment by guedou** [comment]
@haim0n cool !

I decided that March is PR-month, and started to do some cleaning =)

**Pull request merged**